### ISSUE TITLE
Hantera mycket data

### ISSUE THREAD

**Issue by MalinAhlberg** [majsan/tolkordlistor#15]
Hur hantera stora sökningar?

Tänkbara lösningar:
- man får vänta
- visa bara topp X
- pagination
- cachning

**Comment by MalinAhlberg** [comment]
Bra sak att cacha: tom sökning för varje sökområde, eftersom detta händer ofta. Kanske även kombinationer.

**Comment by majsan** [comment]
Gränssnittet låser sig lite när den måste rendrera långa listor. Det är alltså inte så att det låser sig under själva anropet, utan när anropet har svarat. Möjligtvis så kan förbättra performance genom att göra så lite som möjligt per ingång i frontenden, men grunden till problemet är för mycket data.

**Comment by MalinAhlberg** [comment]
Vi kör på att bara visa A-ord för långa sakområden tills vidare.

**Comment by MalinAhlberg** [comment]
- `max_size` sätts ganska lågt (500)
- gränssnittet kommunicerar till användaren om alla ord inte visas
- spänstigare lösning med till exempel pagination är för tidskrävande för detta projekt

**Comment by Gurras** [comment]
Att bara visa A-ord för långa områden är väl ok för "tomma sökningar", tror jag. Men att samma beteende gäller även för HTML-exporten är väldigt förvirrande och inte särskilt användbart.

**Comment by Gurras** [comment]
Hur öppnar man man ett stängt ärende? Det kanske inte jag kan göra?

**Comment by MalinAhlberg** [comment]
> Hur öppnar man man ett stängt ärende? Det kanske inte jag kan göra?

Det finns en reopen-knapp här nedanför, men du kanske inte får trycka på den för github. Jag gör det nu.

> Att bara visa A-ord för långa områden är väl ok för "tomma sökningar", tror jag. Men att samma beteende gäller även för HTML-exporten är väldigt förvirrande och inte särskilt användbart.

Ok, vi funderade lite på vad som var bäst här. Vill ni alltså ha att hela sökningen tas med i exporten, oavsett storlek och vad söksidan visar?

**Comment by Gurras** [comment]
Ja, annars har man väl inte särskilt stor nytta av exporten. Vi tror väl också att exporten (åtminstone till en början) inte kommer vara särskilt mycket använd. Och om den blir det, så är det väl inget problem att användaren får vänta lite, i synnerhet inte om den kan få nånslags "Vänta lite, jag jobbar"-flash. Emot talar egentligen bara eventuella serverkapacitetsproblem.

**Comment by evali1** [comment]
Det verkade inte finnas något sätt att välja 'full export' när bara A laddas! Och det bör man ju kunna. 

Vet inte om det skulle funka att få text längst ner, t.ex. 'Ladda hela', resp (kanske) 'Ladda bokstav...' och en liten ruta - ifall nu inte folk begriper att det är samma sak som att göra sökning med 'börjar på ...'; Möjligen motsvarande när man begär HTML-export?

Vi noterade också att det blir långt att skrolla till exportknappen tom när bara A laddas. Men man vill ju inte ha den upptill, och såna där små hover-menyer som ska ligga i fönstrets nederkant brukar vara rätt glitchiga, hoppiga och störande. Så jag vet inte hur man skulle lösa det.

/Eva (som ska försöka skärpa sig och skriva här i stället....)

**Comment by MalinAhlberg** [comment]
Vårt främsta/enda motargument var att det skulle bli ointuitivt att inte få samma resultat i webgränssnittet som i exporten. Men det verkar ju också rimligt att kunna få alla ord, så vi gör så från och med nu.

Pga av mobilers obraighet är det dumt att rakt av ge möjligheten till att ladda för mycket, om man inte smartar sig och samtidigt döljer/tar bort något annat. Vi planerar dock att ha en liten text längst ner i de fall de hela ens sökning inte kan visas (sista checkboxen på den här issuen https://github.com/majsan/tolkordlistor/issues/36). Tror ni det blir bra (nog)?

Jag gör en ny issue för var exportknappen ska visas https://github.com/majsan/tolkordlistor/issues/52

**Comment by MalinAhlberg** [comment]
@Gurras @majsan @evali1 Kan vi stänga den här? I annat fall sätta den som `future`?
/mvh issue-städaren

**Comment by Gurras** [comment]
Future blir nog bra. Denna tråd kan väl vara bra att ha tillgång till om man börjar fundera på problemet med mkt data igen.

**Comment by evali1** [comment]
Jo, fast ... man har väl tillgång till stängda ämnen också? Ser så ut när
github föreslår sökresultat iaf. Och däri finns ju ockå en massa som man
isf vill ha tillgång till!

Så om det är som jag tror så tycker jag att om inte @Gurras
https://github.com/Gurras protesterar väldans högljutt så kan man stänga
(="arkivera"??) för att städa upp i aktuella issues!
